pkg/apis/configuration/validation/dos.go

```diff
@@ -33,26 +32,26 @@ func ValidateAppProtectDosLogConf(logConf *unstructured.Unstructured) error {
 	return nil
 }
 
-var accessLog = regexp.MustCompile(`^(((\d{1,3}\.){3}\d{1,3}):\d{1,5})$`)
+var (
+	dosLogDstEx = regexp.MustCompile(`(\S+:\d{1,5})|stderr`)
+)
```

Comment:
yes, if we allow dns names, they can almost be any string. Is there some validation you think we can do?

Reply:
Suggestion:
use the format of ip=191.2.2.2 or dns=dns.bla.bla
in dos.go:

> var (
> 	dosLogDstEx = regexp.MustCompile(`(\S+:\d{1,5})|stderr`)
> )
> 
> // ValidateAppProtectDosLogDest validates destination for log configuration
> func ValidateAppProtectDosLogDest(dstAntn string) error {
> 	errormsg := "Error parsing App Protect Dos Log config: Destination must follow format: <ip=ip-address | localhost | dns=dns name>:<port> or stderr"
> 	if !dosLogDstEx.MatchString(dstAntn) {
> 		return fmt.Errorf("%s Log Destination did not follow format", errormsg)
> 	}
> 	if dstAntn == "stderr" {
> 		return nil
> 	}
> 
> 	dstchunks := strings.Split(dstAntn, ":")
> 
> 	// // This error can be ignored since the regex check ensures this string will be parsable
> 	port, _ := strconv.Atoi(dstchunks[1])
> 	if port > 65535 || port < 1 {
> 		return fmt.Errorf("error parsing port: %v not a valid port number", port)
> 	}
> 
> 	dstchunks = strings.Split(dstchunks[0], "=")
> 
>     if dstchunks[0] == "ip" {
>         if net.ParseIP(dstchunks[1]) == nil {
>             return fmt.Errorf("Error parsing host: %v is not a valid ip address", dstchunks[0])
>         }
>     }
> 
> 	return nil
> }

in configuration.go:

> func generateDosLogDest(dest string) string {
> 	if dest == "stderr" {
> 		return dest
> 	}
> 	if strings.HasPrefix(dest, "ip=") || strings.HasPrefix(dest, "dns=") {
> 		dstchunks := strings.Split(dest, "=")
> 		return dstchunks[1]
> 	}
> 	return "syslog:server=" + dest
> }